### PR TITLE
Avoid generating `.seed` files

### PR DESCRIPTION
**Pull request by rvhonorato**
You are about to submit a new Pull Request. Before continuing make sure you read the [contributing guidelines](CONTRIBUTING.md) and that you comply with the following criteria:

- [x] You have sticked to Python. Please talk to us before adding other programming languages to HADDOCK3
- [x] Your PR is about CNS
- [x] Your code is well documented: proper docstrings and explanatory comments for those tricky parts
- [x] You structured the code into small functions as much as possible. You can use classes if there is a (state) purpose
- [x] Your code follows our coding style
- [x] You wrote tests for the new code
- [x] `tox` tests pass. *Run `tox` command inside the repository folder*
- [ ] `-test.cfg` examples execute without errors. *Inside `examples/` run `python run_tests.py -b`*
- [x] PR does not add any dependencies, unless permission granted by the HADDOCK team
- [x] PR does not break licensing
- [x] Your PR is about writing documentation for already existing code :fire:
- [x] Your PR is about writing tests for already existing code :godmode:

---

This PR adds a `seed` property to `libontology.PDBFile` and refactors `libcns` to use this seed when creating the `.inp` files/strings. I also added several tests that were missing for both `libontology` and `libcns`.

I have also changed the `.cns` files that would create this `.seed` files.

In summary what happens is; in `rigidbody` the seed is set according to:

```python
seed = self.params["iniseed"] * idx
```

This is passed to `libcns.prepare_cns_input` which will create a "eval line" with this value.

This seed number is also added to the _expected output_, and subsequent modules will either read this seed - if present, or generate a random one. This should ensure compatibility and not break the current workflows.

@amjjbonvin I have found the place in the CNS routines that would write this `.seed` file but I didn't find the places where it would be read - could you please check if I missed something?

### CODE REVIEW

**Review by VGPReys — APPROVED** [review]
Easy solution to replace one file by a single line !
Also simplify the CNS code

**Comment by rvhonorato** [comment]
Oh yeah, the place where it would read the `.seed` is part of the removed block:

```fortran
    if ($result eq true) then
        @@$fileseed (seed=$seed;npart=$npart )
    else
        evaluate ($seed = $count*$Saprotocol.iniseed)
    end if
```

**Review by mgiulini — APPROVED** [review]

**Review by amjjbonvin — APPROVED** [review]
A nice one - all examples giving consistent results.
